packages/client/tests/functional/field-reference/json/tests.ts

```diff
@@ -0,0 +1,131 @@
+// @ts-ignore
+import type { PrismaClient } from '@prisma/client'
+
+import testMatrix from './_matrix'
+
+declare let prisma: PrismaClient
+
+testMatrix.setupTestSuite(
+  ({ provider }) => {
+    afterEach(async () => {
+      await prisma.product.deleteMany({})
+    })
+
+    test('simple equality', async () => {
+      await prisma.product.createMany({
+        data: [
+          {
+            title: 'Potato',
+            properties1: {
+              kind: 'root vegetable',
+            },
+            properties2: {
+              kind: 'root vegetable',
+            },
+          },
+
+          {
+            title: 'Apple',
+            properties1: {
+              color: 'red',
+            },
+            properties2: {
+              kind: 'fruit',
+            },
+          },
+        ],
+      })
+
+      const products = await prisma.product.findMany({
+        where: { properties1: { equals: prisma.product.fields.properties2 } },
+      })
+
+      expect(products).toEqual([expect.objectContaining({ title: 'Potato' })])
+    })
+
+    test('does not conflict with {_ref: "something"} json value', async () => {
```

Comment:
What does this mean, and whats the reason for this edge case?

Reply:
When converting to GraphQL, field reference turns into `{ _ref: "columnName" }` object.

So this query:

```ts
where: { field: { gt: prisma.model.fields.someField }}
```

becomes this:

```query
where: { field: { gt: { _ref: 'someField' }}}
```

In case of JSON column, we need to ensure that if user explicitly passes `{ _ref: 'something' }` as an input to json filter it is treated as a raw json value and not as a field reference.